source/addonStore/dataManager.py

```diff
@@ -273,20 +280,42 @@ def getLatestAddons(
 					cachedLanguage=self._lang,
 					nvdaAPIVersion=_LATEST_API_VER,
 				)
-			elif onDisplayableError is not None:
-				from gui.message import DisplayableError
-
-				displayableError = DisplayableError(
+			else:
+				self._do_displayError(
+					onDisplayableError,
 					# Translators: A message shown when fetching add-on data from the store fails
 					pgettext("addonStore", "Unable to fetch latest add-on data for incompatible add-ons."),
-					self._updateFailureMessage,
 				)
-				callLater(delay=0, callable=onDisplayableError.notify, displayableError=displayableError)
 
 		if self._latestAddonCache is None:
 			return _createAddonGUICollection()
 		return deepcopy(self._latestAddonCache.cachedAddonData)
 
+	def _do_displayError(
+		self,
+		onDisplayableError: "DisplayableError.OnDisplayableErrorT | None",
+		displayMessage: str,
+		titleMessage: str | None = _updateFailureMessage,
+		showTip: bool = True,
+	):
+		"""Display a DisplayableMessage if an OnDisplayableError action is given.
+
+		See gui.message.DisplayableError for further information.
+
+		:param onDisplayableError: The displayable error action.
+		:param displayMessage: Body of the displayable error.
+		:param titleMessage: Title of the displayable error, defaults to _updateFailureMessage.
+		:param showTip: Whether or not to show a suggestion of what to try, defaults to True.
```

Comment:
is there any cases where this would be False?

Reply:
I can't think of any, but thought it best to include the option just in case. Happy to remove if you have a reason not to include the option though.

Reply:
Yeah I think it's better to avoid complexity unless a clear usage exists/is expected